### PR TITLE
Signed SAML response to SP

### PR DESCRIPTION
**Pull request by Zogoo**
This pull request contains 2 changes.
1. Signed response
Currently, gem only signing Assertion, but there are several patterns can be exists ([here is examples](https://www.samltool.com/generic_sso_res.php)) one of them is signed response.
2. Travis build fixes for Ruby and Rails version dependency issue

### CODE REVIEW

**Comment by kis-xt** [comment]
Twilio SSO via SAML doesn't work without Response > Signature element, this patch fix it

```
{"code": 70004, "detail": "Your AccountSid or AuthToken was incorrect.", 
"message": "authentication failed: The Message of the Response is not signed and the SP requires it", 
"more_info": "https://www.twilio.com/docs/errors/70004", "status": 401}
```

**Comment by Zogoo** [comment]
@kis-xt

Thanks for feedback, you are becoming my first reviewer my pull requests. :)

If you can show me part of your source code which setting up signed_response_opts flag true, that will be good help for me.

So, I only tried my fixes with GSuite SSO app, it was working well. But let me try it on my side with Twillio SSO. 

Thanks

**Comment by kis-xt** [comment]
@Zogoo 
```
  def idp_make_saml_response(found_user) # not using params intentionally
    encode_response(found_user, signed_message: true)
  end
```

**Comment by jphenow** [comment]
I'll take another look at this shortly. I think we can merge and move forward though. Just want to be sure I understand if there breaking additions etc. 

Thanks for all the work @Zogoo

**Comment by Zogoo** [comment]
> I'll take another look at this shortly. I think we can merge and move forward though. Just want to be sure I understand if there breaking additions etc.
> 
> Thanks for all the work @Zogoo

You might wonder @kis-xt comment, I still did not check Twilio SSO, but in my side, it's working with Google and AWS which are required singed response.  And test I've already confirmed that signature of the response is correct with OneLogin::RubySaml::Response.

File: lib/saml_idp/response_builder.rb [hunk 1]
```diff
@@ -52,6 +64,7 @@ def build
     def response_id_string
       "_#{response_id}"
     end
+    alias_method :reference_id, :response_id
```
Comment (jphenow) [inline lib/saml_idp/response_builder.rb]: think we can put this up closer to the attr-accessor

Comment (Zogoo) [reply lib/saml_idp/response_builder.rb]: I have done with changes as you requested. Could you please take a look at it. My English not perfect, feel free if you want to change my explanation of README.

**Comment by chienkira** [comment]
Hi. I would like to ask when v0.10.0 is released?

File: README.md [hunk 2]
```diff
@@ -72,6 +72,22 @@ end
 
 ## Configuration
 
+#### Singed assertions and Signed Response
```
Comment (mjobin-mdsol) [inline README.md:75]: typo, `Signed`

File: README.md [hunk 3]
```diff
@@ -72,6 +72,22 @@ end
...
+
+By default SAML Assertion will be signed with an algorithm which defined to `config.algorithm`. Because SAML assertions contain secure information used for authentication such as NameID.
```
Comment (mjobin-mdsol) [inline README.md:77]: would it make sense to replace
> which defined to
by
> which is defined by
??

File: README.md [hunk 4]
```diff
@@ -72,6 +72,22 @@ end
...
+
+Signing SAML Response is optional, but some security perspective SP services might require Response message itself must be signed.
+For that, you can enable it with `config.signed_message` option. [More about SAML spec](https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf#page=68)
+
+#### Signing algorithm
+
+Following algorithms you can set in your response signing algorithm
+:sha1 - RSA-SHA1 default value but not recommended to production environment
```
Comment (mjobin-mdsol) [inline README.md:85]: why keeping `sha1` default and simply not removing it ?

any reason not to use `sha512` as default ?

Comment (Zogoo) [reply README.md:85]: Sha1 is used for this gem signature feature. I don't want to start to change it when I reuse it.

Comment (mjobin-mdsol) [reply README.md:85]: ok

**Comment by mjobin-mdsol** [comment]
@jphenow any plan on making a release with these changes?

**Comment by jphenow** [comment]
Much of [this](https://github.com/saml-idp/saml_idp/issues/125#issuecomment-575668016) still holds true and may be even more true now. I will continue to do my best to periodically be a steward of the work folks submit but the reality is that this far from even my top 10 of priorities of the last several years. Apologies if there's an expectation or hope to see more out of this. I encourage folks to continue to branch for their own needs and I will periodically do my best to merge here. I'm open to allowing others to take the reigns if there's ample interest in executing on that. I had a less than stellar experience with the last time someone attempted to take over because they were less engaged though seemed very eager to slap their name all over the project. But @Zogoo has done a lot of great work and I just haven't been able to keep this project as much in mind as I'd like to. If @Zogoo or others would like some elevated privileges here I'm willing to begin some handover or shared responsibility where there's interest.

Anywho - sorry for long description, and late response. The PR is merged and v0.10.0 is out.

**Pull request merged**